pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -1262,8 +1262,21 @@ func (l *LibvirtDomainManager) FreezeVMI(vmi *v1.VirtualMachineInstance, unfreez
 func (l *LibvirtDomainManager) UnfreezeVMI(vmi *v1.VirtualMachineInstance) error {
 	l.cancelSafetyUnfreeze()
 	domainName := api.VMINamespaceKeyFunc(vmi)
-	// fs thaw is idempotent by itself
-	_, err := l.virConn.QemuAgentCommand(`{"execute":"guest-fsfreeze-thaw"}`, domainName)
+	cmdResult, err := l.virConn.QemuAgentCommand(`{"execute":"`+string(agentpoller.GET_FSFREEZE_STATUS)+`"}`, domainName)
+	if err != nil {
+		log.Log.Errorf("Failed to get status before unfreeze vmi, %s", err.Error())
+		return err
+	}
+	fsfreezeStatus, err := agentpoller.ParseFSFreezeStatus(cmdResult)
+	if err != nil {
+		log.Log.Errorf("Failed to parse status before unfreeze vmi, %s", err.Error())
+		return err
+	}
```

Comment:
I think we should skip the thaw only if we positively can confirm it is not frozen.  In the case where there is an error with fs-status then I think we should still call thaw just to be safe.

Reply:
I updated the code